numpy/f2py/crackfortran.py

```diff
@@ -2323,7 +2323,7 @@ def _get_depend_dict(name, vars, deps):
 
         if '=' in vars[name] and not isstring(vars[name]):
             for word in word_pattern.findall(vars[name]['=']):
-                if word not in words and word in vars:
+                if word not in words and word in vars and word != name:
                     words.append(word)
```

Comment:
Can you add a comment here somewhere indicating that `word` may not be a real variable name?

Reply:
Sure, I've just updated the MR, adding this comment:
                # The word_pattern may return values that are not
                # only variables, they can be string content for instance